api/src/opentrons/protocol_engine/execution/equipment.py

```
@@ -161,9 +162,17 @@ async def load_pipette(
         other_mount = mount.other_mount()
         other_pipette = self._state_store.pipettes.get_by_mount(other_mount)
 
-        cache_request = {mount.to_hw_mount(): pipette_name.value}
+        cache_request = {
+            mount.to_hw_mount(): pipette_name.value
+            if isinstance(pipette_name, PipetteNameType)
+            else pipette_name
+        }
         if other_pipette is not None:
-            cache_request[other_mount.to_hw_mount()] = other_pipette.pipetteName.value
+            cache_request[other_mount.to_hw_mount()] = (
```

Shouldn't we raise an error here rather than modify the request to be a 96 channel? This behavior seems confusing/unexpected otherwise.

---

nice catch! I agree! if the other pipette is set and we are loading a 96 channel pipette we should probably raise an exception

---

Can we punt this to a TODO? This will not be our production 96 channel logic, and the only reason we have to pass in whatever else is attached is because the hardware API doesn't give us a better API than `cache_instruments` to say say "verify that this pipette is attached to this mount"

---

added a todo for now

---

@mcous is whatever the protocol engine needs from the hardware controller for this ticketed somewhere?